src/nvim/eval.c

```diff
@@ -22483,10 +22491,10 @@ bool eval_has_provider(const char *name)
     } \
   }
 
-  static int has_clipboard = -1;
-  static int has_python = -1;
-  static int has_python3 = -1;
-  static int has_ruby = -1;
+  bool has_clipboard = false;
+  bool has_python = false;
+  bool has_python3 = false;
+  bool has_ruby = false;
```

Comment:
Can't do this. Old way avoided recursion.

Reply:
Oops, right. I add back the `static` but keep the type change considering that the function is supposed to return `bool` anyway.

Reply:
bool is not going to work, the type must be a ternary.

Reply:
Like the `TriState` enum?  Maybe that will make this more obvious for the next person that looks at the code.

Reply:
Argh, sorry. I'm pretty confused today. I now see that it wouldn't work, because it would never take the `if` path. I ditch the commit.